### ISSUE TITLE
[Bug] Cannot build Spleeter with GCC 10 due to outdated dependencies (scipy 1.4.1)

### ISSUE THREAD

**Issue by Sophira** [deezer/spleeter#649]
- [x] I didn't find a similar issue already open.
- [x] I read the documentation (README AND Wiki)
- [x] I have installed FFMpeg
- [x] My problem is related to Spleeter only, not a derivative product (such as Webapplication, or GUI provided by others)

## Description

When trying to build Spleeter's most recent commit (efdd159e) into a clean virtualenv using Python 3.9, GCC 10, and Poetry 1.1.8, the installation of scipy 1.4.1 fails, I believe due to https://github.com/scipy/scipy/issues/11611.

This means that it's currently impossible to build Spleeter under GCC 10 as it specifically requires version 1.4.1 of scipy in its `poetry.lock` file.

## Step to reproduce

1. `virtualenv -p 3.9 venv`
2. `source venv/bin/activate`
3. `pip install --upgrade pip`
4. `pip install poetry`
5. Change directory to the Spleeter git tree
6. `poetry install`

## Output

```
[...]
  • Installing scipy (1.4.1): Failed
[...]
    scipy/sparse/linalg/eigen/arpack/ARPACK/SRC/cnaup2.f:503:35:

      392 |             call ivout (logfil, 1, iter, ndigit,
          |                                   2
    ......
      503 |             call ivout (logfil, 3, kp, ndigit,
          |                                   1
    Error: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
    scipy/sparse/linalg/eigen/arpack/ARPACK/SRC/cnaup2.f:666:38:

      392 |             call ivout (logfil, 1, iter, ndigit,
          |                                   2
    ......
      666 |                call ivout (logfil, 2, kp, ndigit,
          |                                      1
    Error: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
[...]
    error: Command "/usr/bin/gfortran -Wall -g -ffixed-form -fno-second-underscore -fPIC -O3 -funroll-loops -Iscipy/sparse/linalg/eigen/arpack/ARPACK/SRC -I/tmp/pip-build-env-ck87paxy/overlay/lib/python3.9/site-packages/numpy/core/include -c -c scipy/sparse/linalg/eigen/arpack/ARPACK/SRC/cnaup2.f -o build/temp.linux-x86_64-3.9/scipy/sparse/linalg/eigen/arpack/ARPACK/SRC/cnaup2.o" failed with exit status 1
    ----------------------------------------
    ERROR: Failed building wheel for scipy
  Failed to build scipy
  ERROR: Could not build wheels for scipy which use PEP 517 and cannot be installed directly
```

## Environment

|                   |                                 |
| ----------------- | ------------------------------- |
| OS                | Linux |
| Installation type | Poetry  |
| RAM available     | 24 GiB total, 8 GiB available |
| Hardware spec     | CPU: Intel(R) Core(TM) i7-5820K CPU @ 3.30GHz |

(As this is a compilation issue and not a runtime issue, I have not supplied the details of my GPU; if this is needed, please let me know.)

## Additional context

Assuming that I have the right issue linked, then the problem was fixed in scipy some time ago.

**Comment by Faylixe** [comment]
Well, actually we are not supporting Python 3.9 officially anyway as we rely on a TF version that is not 3.9 compliant.  Plus poetry make it hard to infer with pip directive such as pep compliance and `--no-binary` so we will probably wait before upgrading current `scipy` version. Although if you manage to make it work do not hesitate to submit a PR or notify us with your progress on this issues we would gladly review it :)

**Comment by Sophira** [comment]
Yeah, after I submitted this issue, I noticed when I tried using GCC 9 instead that the install was having issues with `pandas`, too. It looks like the latest Python version I could use without errors is 3.8, at which point Poetry just installs a binary anyway and thus bypasses the GCC 10 issue.

I can't promise that I'll be able to work on this, though, sorry! I may give it a shot if I can but it's quite likely that I won't have the time. For now, installing with Python 3.8 is good enough for me, but this does seem like something which will eventually need to be dealt with so if I can I'll take a look, but no promises.

Thank you for the response!
